TestValidateCronJob cant pass in macos

### What happened?

```bash
$ go test ./pkg/apis/batch/validation -run  TestValidateCronJob

--- FAIL: TestValidateCronJob (0.01s)
    validation_test.go:1355: expected failure for spec.timeZone: Invalid value: "AMERICA/new_york": unknown time zone AMERICA/new_york
    validation_test.go:1378: expected failure for spec.timeZone: Invalid value: "AMERICA/new_york": unknown time zone AMERICA/new_york
FAIL
FAIL    k8s.io/kubernetes/pkg/apis/batch/validation     0.028s
FAIL

```

### What did you expect to happen?

test withnot FAIL

### How can we reproduce it (as minimally and precisely as possible)?

In a macos.
do `make test` or `go test ./pkg/apis/batch/validation -run  TestValidateCronJob`

### Anything else we need to know?

https://github.com/golang/go/issues/21512  golang upstrem disscusion.

We use time.LoadLocation to validate time_zone setting in object spec like cronjob.(It is ok)

We also add a test try to validate `AMERICA/new_york` is not a crroct value.

This test is not true sometimes, for example, in default macos this is a true time_zone value.(maybe also true in other case insensitive os, but I have no device to try)

At least we should get the test to pass on macos, I hope.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>


### Cloud provider

<details>

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here

Darwin MacBook-Pro.local 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

```
</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

--- a/pkg/apis/batch/validation/validation_test.go
+++ b/pkg/apis/batch/validation/validation_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package validation
 
 import (
+	"runtime"
 	"strings"
 	"testing"
 
@@ -1033,23 +1034,6 @@ func TestValidateCronJob(t *testing.T) {
 				},
 			},
 		},
-		"spec.timeZone: Invalid value: \"AMERICA/new_york\": unknown time zone AMERICA/new_york": {
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "mycronjob",
-				Namespace: metav1.NamespaceDefault,
-				UID:       types.UID("1a2b3c"),
-			},
-			Spec: batch.CronJobSpec{
-				Schedule:          "0 * * * *",
-				TimeZone:          &timeZoneBadCasing,
-				ConcurrencyPolicy: batch.AllowConcurrent,
-				JobTemplate: batch.JobTemplateSpec{
-					Spec: batch.JobSpec{
-						Template: validPodTemplateSpec,
-					},
-				},
-			},
-		},
 		"spec.timeZone: Invalid value: \" America/New_York\": unknown time zone  America/New_York": {
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      "mycronjob",
@@ -1347,6 +1331,26 @@ func TestValidateCronJob(t *testing.T) {
 				},
 			},
 		},
+	}
+	if runtime.GOOS != "darwin" {
+		// Skip this error case on darwin, see https://github.com/golang/go/issues/21512
+		errorCases["spec.timeZone: Invalid value: \"AMERICA/new_york\": unknown time zone AMERICA/new_york"] = batch.CronJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "mycronjob",
+				Namespace: metav1.NamespaceDefault,
+				UID:       types.UID("1a2b3c"),
+			},
+			Spec: batch.CronJobSpec{
+				Schedule:          "0 * * * *",
+				TimeZone:          &timeZoneBadCasing,
+				ConcurrencyPolicy: batch.AllowConcurrent,
+				JobTemplate: batch.JobTemplateSpec{
+					Spec: batch.JobSpec{
+						Template: validPodTemplateSpec,
+					},
+				},
+			},
+		}
 	}
 
 	for k, v := range errorCases {

